### DISCUSSION TITLE
Hard refresh (CTRL + Shift + R) results in 404

### DISCUSSION THREAD

**Discussion by ThomasFindlay** [vercel/community#799] [Help] [answered]
Hey. I seem to have a weird issue with my website. It's a static SPA that I deployed on Vercel. I have added the **vercel.json** file with rewrite config, so all requests are forwarded to the **index.html** file.

```json
{
  "rewrites": [{ "source": "/(.*)", "destination": "/index.html" }]
}

```

A website can be accessed fine when going to any URL. It also works fine if I reload it using CTRL + R, so the rewrite seem to work correctly. However when I do a hard reload with CTRL + SHIFT + R, Vercel returns a 404 not found page. Could anyone shed some light on what could be the reason for that and how it can be fixed?

Here's the URL to the website

https://novenaofpompeii.com/en/

Here's a gif of what happens

![novena-hard-reload](https://user-images.githubusercontent.com/20544841/186647020-9d206207-1323-43d6-882c-d3d913351556.gif)

**Comment by amyegan** [comment, marked as answer]
_Thomas solved this and shared the answer here: [discussions/799#discussioncomment-3478065](https://github.com/orgs/vercel/discussions/799#discussioncomment-3478065)_ 🦸 

Thanks for reaching out @ThomasFindlay! Confirmed I get the 404 error on first visit when I go to https://novenaofpompeii.com/en directly. The site does load if I go to https://novenaofpompeii.com first and let it redirect me, and it works if I refresh or visit the `/en` version afterward. And hard refresh does result in the 404 again for https://novenaofpompeii.com/en

This seems like it could be related to the method you are using to redirect to `/en`. Are you able to share your repo or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)? That would help me identify what's causing this problem for you.

**Reply by ThomasFindlay** [reply]
Thank you for a quick reply @amyegan. Actually, I managed to solve it. However, here's a brief explanation of the problem and the solution for anyone who runs into this problem in the future.

I use pnpm workspaces for the project, and it contains 3 workspaces - client, server and shared. The client workspace is the SPA deployed on Vercel. Initially, I had the vercel.json file with the rewrites config inside of that SPA workspace. However, it turns out Vercel did not pick it up. Instead, I had to move it to the root directory. This solved the issue.

**Comment by alfiomy** [comment]
Good
